internal/store/store.go

```diff
@@ -147,28 +159,31 @@ type CacheStores struct {
 	l *sync.RWMutex
 }
 
+var s Store
+
 // NewCacheStores is a convenience function for CacheStores to initialize all attributes with new cache stores
-func NewCacheStores() (c CacheStores) {
-	c.ClusterPlugin = cache.NewStore(clusterResourceKeyFunc)
-	c.Consumer = cache.NewStore(keyFunc)
-	c.Endpoint = cache.NewStore(keyFunc)
-	c.IngressV1 = cache.NewStore(keyFunc)
-	c.IngressClassV1 = cache.NewStore(clusterResourceKeyFunc)
-	c.IngressV1beta1 = cache.NewStore(keyFunc)
-	c.HTTPRoute = cache.NewStore(keyFunc)
-	c.UDPRoute = cache.NewStore(keyFunc)
-	c.TCPRoute = cache.NewStore(keyFunc)
-	c.TLSRoute = cache.NewStore(keyFunc)
-	c.ReferencePolicy = cache.NewStore(keyFunc)
-	c.KnativeIngress = cache.NewStore(keyFunc)
-	c.Plugin = cache.NewStore(keyFunc)
-	c.Secret = cache.NewStore(keyFunc)
-	c.Service = cache.NewStore(keyFunc)
-	c.TCPIngress = cache.NewStore(keyFunc)
-	c.UDPIngress = cache.NewStore(keyFunc)
-	c.KongIngress = cache.NewStore(keyFunc)
-	c.l = &sync.RWMutex{}
-	return
+func NewCacheStores() CacheStores {
+	return CacheStores{
+		IngressV1beta1:  cache.NewStore(keyFunc),
+		IngressV1:       cache.NewStore(keyFunc),
+		IngressClassV1:  cache.NewStore(clusterResourceKeyFunc),
+		Service:         cache.NewStore(keyFunc),
+		Secret:          cache.NewStore(keyFunc),
+		Endpoint:        cache.NewStore(keyFunc),
+		HTTPRoute:       cache.NewStore(keyFunc),
+		UDPRoute:        cache.NewStore(keyFunc),
+		TCPRoute:        cache.NewStore(keyFunc),
+		TLSRoute:        cache.NewStore(keyFunc),
+		ReferencePolicy: cache.NewStore(keyFunc),
+		Plugin:          cache.NewStore(keyFunc),
+		ClusterPlugin:   cache.NewStore(clusterResourceKeyFunc),
+		Consumer:        cache.NewStore(keyFunc),
+		KongIngress:     cache.NewStore(keyFunc),
+		TCPIngress:      cache.NewStore(keyFunc),
+		UDPIngress:      cache.NewStore(keyFunc),
+		KnativeIngress:  cache.NewStore(keyFunc),
+		l:               &sync.RWMutex{},
+	}
```

Comment:
Could you please help me to understand the impetus for this change?

Reply:
Yes, sure! This is refactored because we have `CacheStores` to return the values as well, so it is easy to put the things in a more concise way to return wrt. previous return signature. 😄

Reply:
Ok it sounds like the reason was simply for preference, which is fine and it's what I was expecting.